### PR TITLE
Template sweep -- fix bugs in Vue and Lit

### PR DESCRIPTION
**Pull request by pdaoust**
I don't claim this is a thorough sweep that catches all bugs, but it does catch all bugs discovered via following the Getting Started Guide but choosing Lit and Vue instead of Svelte for the templating engine.

### CODE REVIEW

**Comment by ThetaSinner** [comment]
@pdaoust Same with this one, please can it target `main`? Then would you be willing to go through and comment on the PR with a description of what is being changed/fixed so I can do some testing?

I'm happy reviewing the code but I don't know the repo well enough to understand what's changing without having a conversation about it if that's okay! :)

**Comment by ThetaSinner** [comment]
This is now ready for review and can be merged when ready

File: src/scaffold/entry_type.rs [hunk 1]
```diff
@@ -101,6 +101,15 @@ pub fn scaffold_entry_type(
...
     };
 
+    let widget_fields = fields
+        .iter()
+        .map(|f| f.clone())
+        .filter(|f| match f.widget {
+            Some(_) => true,
+            None => false
+        })
+        .collect();
```
Comment (c12i) [inline src/scaffold/entry_type.rs:111]: A little more cleaner 🙂

```suggestion
    let widget_fields = fields
        .iter()
        .cloned()
        .filter(|f| f.widget.is_some())
        .collect();
```

**Review by c12i — CHANGES_REQUESTED** [review]
Discovered similar bugs, at least for the `Vue` and `Svelte` templates hence why I am returning to this PR, I think we should be getting this PR merged as it fixes some critical bugs

@pdaoust kindly let me know if you are in a position to resolve the merge conflicts

File: src/scaffold/entry_type.rs (hunk 1 above)
Comment (c12i) [reply src/scaffold/entry_type.rs:111]: Taking back this comment, perhaps we won't need this fix if the changes #128 are approved

**Comment by c12i** [comment]
@pdaoust, allow me to close this PR since it is stale at the moment and most of the changes made here have long been addressed

**Comment by pdaoust** [comment]
Sure @c12i . Are there any fixes that haven't already been made? Should I do a new PR with those?

**Comment by c12i** [comment]
Thanks @pdaoust, I don't think there are any changes at the moment, I inadvertently addressed them while transitioning the templates to the new styling.

**Pull request closed without merging**